pl_examples/hydra_examples/conf/callbacks/basic.yaml

```diff
@@ -0,0 +1,13 @@
+# @package _group_
+
+functions:
+  print:
+    target: pl_examples.hydra_examples.user_config.MyPrintingCallback
+  message:
+    target: pl_examples.hydra_examples.user_config.MessageCallback
+    params:
+      iter_num: 12
+
+callbacks_list:
+  - ${callbacks.functions.print}
+  - ${callbacks.functions.message}
```

Comment:
Since we are not using composition here it will be simpler to just inline the callbacks:

```suggestion
# @package _group_
callbacks:
  - target: pl_examples.hydra_examples.user_config.MyPrintingCallback
  - target: pl_examples.hydra_examples.user_config.MessageCallback
    params:
      iter_num: 12
```

Reply:
We may not want to use composition here, but we should make things more explicit to allow the user to use composition upon this quite easily...

Reply:
OmegaConf replaces lists completely during merge:
```python
c1 = OmegaConf.create({"list": [1,2,3]})
c2 = OmegaConf.create({"list": [4,5,6]})
c3 = OmegaConf.merge(c1,c2)
assert c3 == {"list": [4,5,6]}
```
One way to compose list element is what this looked like before:

```python
e1 = {"elements": {"e1": {"foo":"bar"}}}
e2 = {"elements": {"e2": {"zoo":"var"}}}
l1 = {"list": ["${elements.e1}", "${elements.e2}"]}
cfg = OmegaConf.merge(l1, e1, e2)
print(cfg.pretty(resolve=True))
```
outputs:
```yaml
list:
- foo: bar
- zoo: var
elements:
  e1:
    foo: bar
  e2:
    zoo: var
```
This lets you reuse e1 and e2 in different lists without repeating them.
This is a bit advanced, and for the sake of this example I think we can just keep things simple.